src/pages/Variants.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useLayoutEffect, useState } from 'react';
+import { useLocation, useParams } from 'react-router-dom';
+import { VUE } from '../model/VUE';
+import { fetchVueData } from '../utils/VUEDataFetcherUtils';
+
+export const Variants: React.FC = () => {
+    const state = useLocation().state;
+    const [vue, setVue] = useState<VUE | undefined>(state);
+
+    const gene = useParams().gene;
+
+    useEffect(() => {
+        if (!state) {
+            const fetchData = async () => {
+                const data = await fetchVueData();
+                setVue(data.find(i => i.hugoGeneSymbol === gene));
+            };
+    
+            fetchData();
+        }
+    }, [gene, state]);
+
+    useLayoutEffect(() => {
+        window.scrollTo(0, 0);
+    })
+
+    if (vue && vue.revisedProteinEffects.length > 1) {
+        const DisplayData = 
+            vue.revisedProteinEffects.map((i) => {
+                return (
+                    <tr>
+                        <td>{i.variant}</td>
+                        <td>{i.transcriptId}</td>
+                        <td>{i.vepPredictedProteinEffect}</td>
+                        <td>{i.vepPredictedVariantClassification}</td>
+                        <td>{i.revisedProteinEffect}</td>
+                        <td>{i.revisedVariantClassification}</td>
+                    </tr>
+                );
+            })
+
+        const references: { referenceText: string, pubmedId: number }[] = [];
+        vue.revisedProteinEffects.map(v => ({ referenceText: v.referenceText, pubmedId: v.pubmedId })).filter(r => {
```

Comment:
Looks like `map` and `filter` are duplicated here because we don't do any actual filter here, maybe we can remove `filter` and use only `map`